client/src/components/web-extension/NFTs/NFTDetails/index.jsx

```diff
@@ -1,16 +1,30 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useLocation } from 'react-router-dom';
+import SendIcon from '@cd/assets/image/send-icon.svg';
 import nftHeaderImage from '@cd/assets/image/nft-header.png';
-import { MiddleTruncatedText } from '../../../Common/MiddleTruncatedText';
-import Copy from '../../../Common/Button/Copy';
-import NoData from '../../../Common/NoData';
+import { NFTAttributes } from './components/NFTAttributes';
+import { NFTTransferForm } from './components/NFTTransferForm';
+
 import './NFTDetails.scss';
 
+const TYPES = {
+	DETAIL: 'DETAIL',
+	TRANSFER_FORM: 'TRANSFER_FORM',
+}
+
 export const NFTDetails = () => {
 	const {
 		state: { nftDetails },
 	} = useLocation();
-	const { image, nftName, nftContractName, metadata, contractAddress, totalSupply } = nftDetails || {};
+	const { image, nftName } = nftDetails || {};
+	const [displayType, setDisplayType] = useState(TYPES.DETAIL);
+ 
+	const onSendClick = () => {
+		setDisplayType(TYPES.TRANSFER_FORM);
```

Comment:
Should strictly check if current `displayType` is not TYPES.TRANSFER_FORM, otherwise it'd trigger re-render under the hood